tests/utilities/test_auto_restart.py

```diff
@@ -720,7 +723,9 @@ def on_train_batch_start(self, trainer, *_) -> None:
                 assert not isinstance(loaders["a"][2].loader.dataset, CaptureMapDataset)
                 assert isinstance(loaders["b"].loader.dataset, RangeIterableDataset)
 
-    with mock.patch.dict(os.environ, {"PL_FAULT_TOLERANT_TRAINING": use_fault_tolerant}):
+    with mock.patch.dict(os.environ, {"PL_FAULT_TOLERANT_TRAINING": use_fault_tolerant}), mock.patch(
+        "pytorch_lightning.trainer.data_loading._validate_fault_tolerant_automatic", lambda x, y: None
```

Comment:
Why is this new mock necessary?

Reply:
It is mocked as the test would fail otherwise has it contains multiple datasets which isn't supported.

This multiple dataset use case is still relevant as the datasets are fully deterministic and this enable us to make sure everything is fully tracked properly internally which worrying about the global random seed overlapping.